### PR TITLE
OGM-1013

### PR DESCRIPTION
**Pull request by schernolyas**
fix issue OGM-1013

### CODE REVIEW

**Comment by Sanne** [comment]
ok to test

**Comment by DavideD** [comment]
Thank @schernolyas , there are some checkstyle problems that make our build fails, could you fix them? 
```
/mnt/jenkins-workdir/workspace/hibernate-ogm-PR/core/src/main/java/org/hibernate/ogm/util/impl/ArrayHelper.java:26: error: Line has leading space characters; indentation should be performed with tabs only.
/mnt/jenkins-workdir/workspace/hibernate-ogm-PR/core/src/main/java/org/hibernate/ogm/util/impl/ArrayHelper.java:27: error: Line has leading space characters; indentation should be performed with tabs only.
/mnt/jenkins-workdir/workspace/hibernate-ogm-PR/core/src/main/java/org/hibernate/ogm/util/impl/ArrayHelper.java:28: error: Line has leading space characters; indentation should be performed with tabs only.
/mnt/jenkins-workdir/workspace/hibernate-ogm-PR/core/src/main/java/org/hibernate/ogm/util/impl/ArrayHelper.java:29: error: Line has leading space characters; indentation should be performed with tabs only.
/mnt/jenkins-workdir/workspace/hibernate-ogm-PR/core/src/main/java/org/hibernate/ogm/util/impl/ArrayHelper.java:30: error: Line has leading space characters; indentation should be performed with tabs only.
/mnt/jenkins-workdir/workspace/hibernate-ogm-PR/core/src/main/java/org/hibernate/ogm/util/impl/ArrayHelper.java:31: error: Line has leading space characters; indentation should be performed with tabs only.
/mnt/jenkins-workdir/workspace/hibernate-ogm-PR/core/src/main/java/org/hibernate/ogm/util/impl/ArrayHelper.java:31: error: White spaces at the end of line
/mnt/jenkins-workdir/workspace/hibernate-ogm-PR/core/src/test/java/org/hibernate/ogm/util/impl/ArrayHelperTest.java:11: error: Using the '.*' form of import should be avoided 
```

File: core/src/test/java/org/hibernate/ogm/util/impl/ArrayHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import static org.junit.Assert.*;
+
+/**
+ * @author Sergey Chernolyas <sergey.chernolyas@gmail.com>
+ */
+public class ArrayHelperTest {
+
+	/**
+	 * Test of toStringArray method, of class ArrayHelper.
+	 */
```
Comment (DavideD) [inline core/src/test/java/org/hibernate/ogm/util/impl/ArrayHelperTest.java]: No need for these comments, they don't add anything to the code

File: core/src/test/java/org/hibernate/ogm/util/impl/ArrayHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+			}
+		}
+	}
+
+	/**
+	 * Test of slice method, of class ArrayHelper.
+	 */
+	@Test
+	public void testSlice() {
+		System.out.println( "slice" );
```
Comment (DavideD) [inline core/src/test/java/org/hibernate/ogm/util/impl/ArrayHelperTest.java]: Can you remove the System.out.? They are unecessary

File: core/src/main/java/org/hibernate/ogm/util/impl/ArrayHelper.java [hunk 3]
```diff
@@ -22,7 +23,13 @@
...
 
 	public static String[][] to2DStringArray(Collection coll) {
-		return (String[][]) coll.toArray( new String[ coll.size() ][] );
+                String[][] arr2D = new String[coll.size()][];
+                Iterator it = coll.iterator();
+                int i = 0;
+                for (;it.hasNext();i++) {
+                    arr2D[i] = new String[]{ (String) it.next()};
+                }                
+		return arr2D;
```
Comment (DavideD) [inline core/src/main/java/org/hibernate/ogm/util/impl/ArrayHelper.java]: Why did you change this?

Comment (schernolyas) [inline core/src/main/java/org/hibernate/ogm/util/impl/ArrayHelper.java]: Because
________________
java.lang.ArrayStoreException: null
        at java.lang.System.arraycopy(Native Method)
        at java.util.Arrays$ArrayList.toArray(Arrays.java:3833)
        at org.hibernate.ogm.util.impl.ArrayHelper.to2DStringArray(ArrayHelper.java:33)
        at org.hibernate.ogm.util.impl.ArrayHelperTest.testTo2DStringArray(ArrayHelperTest.java:35)

File: core/src/test/java/org/hibernate/ogm/util/impl/ArrayHelperTest.java (hunk 2 above)
Comment (schernolyas) [inline core/src/test/java/org/hibernate/ogm/util/impl/ArrayHelperTest.java]: fixed

File: core/src/test/java/org/hibernate/ogm/util/impl/ArrayHelperTest.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+		expectedArr[1] = new String[]{ "2" };
+		String[][] resultArr = ArrayHelper.to2DStringArray( Arrays.asList( new String[]{ "1", "2" } ) );
+		for ( int i = 0; i < resultArr.length; i++ ) {
+			String[] strings = resultArr[i];
+			for ( int j = 0; j < strings.length; j++ ) {
+				String string = strings[j];
+				assertEquals( expectedArr[i][j], string );
+			}
+		}
+	}
```
Comment (DavideD) [inline core/src/test/java/org/hibernate/ogm/util/impl/ArrayHelperTest.java]: I don't think this test is correct, it should be something like:
```
    
import static org.fest.assertions.Assertions.assertThat;
 
	@Test
	public void testTo2DStringArray() {
		List<String[]> keyColumns = new ArrayList<String[]>();
		keyColumns.add( new String[]{"col1", "col2"} );
		keyColumns.add( new String[]{"col3", "col4", "col5"} );
		keyColumns.add( new String[]{"col1"} );

		String[][] expectedArr = new String[3][];
		expectedArr[0] = keyColumns.get( 0 );
		expectedArr[1] = keyColumns.get( 1 );
		expectedArr[2] = keyColumns.get( 2 );

		String[][] resultArr = ArrayHelper.to2DStringArray( keyColumns );
		assertThat( resultArr ).isEqualTo( expectedArr );
	}
```

**Comment by DavideD** [comment]
@schernolyas if you can fix the test, remove the changes to core and sign the CLA (https://cla.jboss.org), I will merge it

Thanks

**Comment by DavideD** [comment]
Can you also squash the commits together?

**Comment by schernolyas** [comment]
changes:
1) squash the commits
2) fix method to2DStringArray and test

**Comment by DavideD** [comment]
Thanks @schernolyas

I've squashed your commits together and sent a new pull request that I'm going to merge.
Your authored commit is now here: https://github.com/hibernate/hibernate-ogm/issues/665

**Comment by vladmihalcea** [comment]
Hi @schernolyas,

We are looking for someone to share their story about the Hibernate OGM in our newsletter. I think this is a good opportunity for your team to advertise your product too. Let me know if you are interested.

Thanks, Vlad

**Comment by schernolyas** [comment]
Hi @vladmihalcea 

About newsletter .... I am  interested. Thank You.  But ...  Can You get some details about the newsletter?

**Comment by vladmihalcea** [comment]
It;s going to be something like [this article](http://in.relation.to/2016/03/07/meet-martin-braun-youngest-hibernate-contributor/). You can send me an email message to my address: vlad at hibernate dot org and I'll send you an interview with 4-5 questions that you can answer them.
